### ISSUE TITLE
Konosuba does not gets scrobled

### ISSUE THREAD

**Issue by justalemon** [sttz/TraktRoller#4]
On S01E01, you get "TypeError: failed to fetch":

![image](https://user-images.githubusercontent.com/11861253/58144009-ceadc780-7c1a-11e9-9659-c1c250c6fb3a.png)

On S01E02, you get "Could not find matching episode on Trakt"

![image](https://user-images.githubusercontent.com/11861253/58144116-2fd59b00-7c1b-11e9-825c-0176e9fbc320.png)

I feel that is related to the matching of Crunchyroll Names with Trakt Show Names, because the name on Crunchyroll is:

```
KONOSUBA -God's blessing on this wonderful world!
```

And on Trakt is:
```
KonoSuba: God's Blessing on This Wonderful World!
```

**Comment by justalemon** [comment]
If the name is causing the problem, a [fuzzy string searching](https://en.wikipedia.org/wiki/Approximate_string_matching) might work by ignoring the casing and removing the special characters (like -, ' and !).

**Comment by sttz** [comment]
Thanks for pointing this out!

This seems to be an issue with Trakt. Trakt has the exact title as an alias but for some reason the show does not appear in the search results. I've opened [an issue](https://github.com/trakt/api-help/issues/76) on Trakt's api-help.

I would prefer not to mess with the title being looked up. The preferred way to resolve weird titles on Crunchyroll would be to add them as alternative titles on TMDB, which then gets picked up by Trakt. I hope Trakt is able to resolve this on their end.

**Comment by justalemon** [comment]
The RegEx `[\+\-&\|!\(\){}\[\]\^"~\*\?:\/]` ([example on RegExr](https://regexr.com/4r7bc)) can be used for matching Solr special characters, with this is possible to get the characters and escape them correctly.

![image](https://user-images.githubusercontent.com/11861253/71349702-675c4b00-254e-11ea-99fe-a4b0de3f25c6.png)

I don't know how RegEx works in JavaScript/TypeScript, so the sintaxis might be wrong or invalid.

**Comment by justalemon** [comment]
Just did a quick check by adding a backslash before - and ! and it works:

![image](https://user-images.githubusercontent.com/11861253/71350605-5a405b80-2550-11ea-8ba3-901e6c439158.png)

The weird thing is, this is not automatically handled by the web page of Trakt.

**Comment by justalemon** [comment]
The other solution would be quoting the title, as mentioned on trakt/api-help#76

**Comment by sttz** [comment]
This has been fixed in 1.0.6.

There are still issues with e.g. S02E11, which gets mistaken for S02E01. But that's a separate issue (E01 has number_abs = 11) and doesn't have a simple fix.
